rewrite-xml/src/main/java/org/openrewrite/xml/XPathMatcher.java

```diff
@@ -92,18 +92,8 @@ public boolean matches(Cursor cursor) {
                     if (index < 0) {
                         return false;
                     }
-                    if (part.startsWith("@")) { // is attribute selector
-                        partWithCondition = part;
-                        tagForCondition = i > 0 ? path.get(i - 1) : path.get(i);
```

Comment:
Using `i` straight up in `path.get()` was completely wrong since `path` is reverted while `parts` is not – `pathIndex` should always be used.

Note that if a part starts with `@`, it MUST be the last part – I wanted to check it here, but then I realized all `if`s actually result in the same code (and conditions matching sub-elements were not handled…).